File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini=INT_MAX;
        int maxprofit=0;
        for(auto &it:prices){
            mini=min(mini,it);
            maxprofit=max(maxprofit,it-mini);
        }
        return maxprofit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int ans=Integer.MIN_VALUE;
        int mini=Integer.MAX_VALUE;
        for(int it:prices){
            mini=Math.min(mini,it);
            ans=Math.max(ans,it-mini);
        }
        return ans;
    }
}